Source/WebCore/Modules/gamepad/GamepadHapticActuator.cpp

```diff
@@ -97,11 +105,12 @@ void GamepadHapticActuator::playEffect(EffectType effectType, GamepadEffectParam
 
     effectParameters.duration = std::min(effectParameters.duration, GamepadEffectParameters::maximumDuration.milliseconds());
 
-    m_playingEffectPromise = WTFMove(promise);
-    GamepadProvider::singleton().playEffect(m_gamepad->index(), m_gamepad->id(), effectType, effectParameters, [this, protectedThis = makePendingActivity(*this), playingEffectPromise = m_playingEffectPromise](bool success) mutable {
-        if (m_playingEffectPromise != playingEffectPromise)
+    currentEffectPromise = WTFMove(promise);
+    GamepadProvider::singleton().playEffect(m_gamepad->index(), m_gamepad->id(), effectType, effectParameters, [this, protectedThis = makePendingActivity(*this), playingEffectPromise = currentEffectPromise, effectType](bool success) mutable {
+        auto& currentEffectPromise = promiseForEffectType(effectType);
+        if (playingEffectPromise != currentEffectPromise)
```

Comment:
What is keeping `currentEffectPromise` alive?

Reply:
promiseForEffectType() returns a reference to a data member that is a RefPtr.